src/DataTable/DataTable.js

```diff
@@ -189,12 +189,23 @@ const DataTable = memo(({
 
   const sortedData = useMemo(() => {
     // server-side sorting bypasses internal sorting
-    if (!sortServer) {
+    if (sortServer) {
+      return data;
+    }
+
+    // use general sorting function when columns has no sort function on it's own
+    const column = sortColumn && columns.find(item => item.selector === sortColumn);
```

Comment:
@ondratra sorry it took me so long to review this!

Overall this looks good, but I am not sure about adding an O(n) lookup on every sort (when `sortDirection` changes). Perhaps there is another way we can refactor this and avoid this extra step? I realize that there are finite columns here and I want to be careful of pre-optimization (perhaps we can measure this to be sure), but sorting is already pretty expensive and I am sensitive to performance issues.